handlers.go

```diff
@@ -4302,3 +4305,53 @@ func (s *Server) handleRunGc(c echo.Context) error {
 
 	return nil
 }
+
+func (s *Server) handleGateway(c echo.Context) error {
+	proto, cc, segs, err := gateway.ParsePath("/" + c.Param("path"))
+	if err != nil {
+		return err
+	}
+
+	redir, err := s.checkGatewayRedirect(proto, cc, segs)
+	if err != nil {
+		return err
+	}
+
+	if redir == "" {
+		s.gwayHandler.ServeHTTP(c.Response().Writer, c.Request())
+		return nil
+	}
+
+	return c.Redirect(302, redir)
+}
+
+const bestGateway = "dweb.link"
+
+func (s *Server) checkGatewayRedirect(proto string, cc cid.Cid, segs []string) (string, error) {
+	if proto != "ipfs" {
+		return fmt.Sprintf("https://%s/%s/%s/%s", bestGateway, proto, cc, strings.Join(segs, "/")), nil
+	}
+
+	var cont Content
+	if err := s.DB.First(&cont, "cid = ? and active and not offloaded", &util.DbCID{cc}).Error; err != nil {
```

Comment:
Is there value in checking for `active`? We are sending hem to the gateway, which will look in the swarm, and likely get back to you with a bitswap session. Might as well serve it if you have it...?
Or does `NOT active` potentially imply you no longer have all the blocks?

Reply:
yeah, not active implies no blocks

Reply:
either no blocks *yet* or no blocks for some other reason (like user deleted the thing)

Reply:
✔️